src/DotLiquid/StandardFilters.cs

```diff
@@ -920,10 +926,12 @@ public static IEnumerable Uniq(object input)
         /// </summary>
         /// <param name="context">The DotLiquid context</param>
         /// <param name="input">Input to be transformed by this filter</param>
-        public static double Abs(Context context, object input)
+        public static object Abs(Context context, object input)
         {
-            Double n;
-            return Double.TryParse(input.ToString(), NumberStyles.Number, context.CurrentCulture, out n) ? Math.Abs(n) : 0;
+            if (decimal.TryParse(input?.ToString(), NumberStyles.Any, context.CurrentCulture, out decimal d))

```

Comment:
double is very different from decimal in its handling of precision error.  Changing between them will generate different results which can be a regression for some customers. This change should only be done behind a version flag to avoid breaking DotLiquid consumers.

Reply:
I had the same concerns, with math functions we did double work, catching an overflow exception before reverting to decimal types.

Reply:
I can get behind a catch exception and recover from error; that should not change output for currently working calls, and we don't need to version-protect valid input failures turned into success.

Reply:
I've added Abs, Ceil and Floor behind a SyntaxCompatibilty flag, and added appropriate tests. It's a lot of work to maintain backwards compatibility! I will have a look at doing the same for the other filters.

Reply:
I've change the behaviour to try to preserve the data type, unless it's a string. If it's a string, it will use the shared coercion to number, and then change the type as determined. 

I can do a similar thing to Round. 

AtLeast and AtMost (and presumably other mathematical functions with 2 arguments) - how do we want to handle these when the inputs are different types? Do we have a preferred order of coercion (e.g. int32, int64, decimal, double)?

Reply:
For all the mathematical functions, when the input types are different, the type with the most range and precision that covers the range and precision of the other type is preferred. And possibly in case of over/under-flow, the type is promoted to one that can handle the range.

Reply:
Here's a double that overflows on conversion to decimal:
https://dotnetfiddle.net/TgoY1q

Reply:
Here's a double that loses precision on conversion to decimal:
https://dotnetfiddle.net/99Gntw

Reply:
Unfortunately, the large number can't be parsed from a string (although I found 1e203 to work). And the precise number happily gets converted to a decimal without loss of precision. Interestingly: The precision loss only looks to be an issue with .NET Core, and not .NET Framework.

I've added all the SyntaxCompatibility flags to maintain old behaviour, and added appropriate tests. I've made AtLeast and AtMost return decimal, which allows all the golden liquid tests to pass. If you think it's worth maintaining the type in these two, it might be better to raise an issue so this can be merged.